encrypt.go

```diff
@@ -152,9 +147,69 @@ func (e *EncryptManager) EncryptCFB(r io.Reader) ([]byte, error) {
 	return encrypted, nil
 }
 
+// EncryptCipherAndNonce is used to encrypt the cipher and nonce generated by AES256-GCM
+// with AES256-CFB using the passphrase provided during instantiation of EncryptManager
+// this allows us to return this information to the user through a more secure manner.
+// this is intended for when AES256-GCM is used to encrypt data within Temporal
+func (e *EncryptManager) encryptCipherAndNonce(key, nonce []byte) ([]byte, error) {
+	// encode nonce
+	encodedNonce := hex.EncodeToString(nonce)
+	// encode cipherkey
+	encodedKey := hex.EncodeToString(key)
+	// format output
+	formattedOutput := fmt.Sprintf("Nonce:\t%s\nCipherKey:\t%s", encodedNonce, encodedKey)
+	// encrypt the output, and return
+	return e.encryptCFB(strings.NewReader(formattedOutput))
+}
+
+// DecryptParams is used to configure decryption for AES256-GCM
+// It expects a hex encoded CipherKey and Nonce
+type DecryptParams struct {
+	CipherKey string
+	Nonce     string
+}
+
+// Decrypt is used to handle decryption of the io.Reader
+// if params is nil, AES256-CFB is assumed
+func (e *EncryptManager) Decrypt(r io.Reader, params *DecryptParams) ([]byte, error) {
+	if params == nil {
```

Comment:
this behaviour is very misleading -  something like `Decrypt(r io.Reader, p Protocol)` where:

```go
type Protocol string
```

and switch